src/features/documentSymbolProvider.ts

```diff
@@ -25,32 +26,23 @@ export default class OmnisharpDocumentSymbolProvider extends AbstractSupport imp
     }
 }
 
-function createSymbols(elements: Structure.CodeElement[], fileName: string): vscode.SymbolInformation[] {
-    let results: vscode.SymbolInformation[] = [];
+function createSymbols(elements: Structure.CodeElement[], filename: string): vscode.DocumentSymbol[] {
+    let results: vscode.DocumentSymbol[] = [];
 
-    Structure.walkCodeElements(elements, (element, parentElement) => {
-        const parentElementName = parentElement ? parentElement.DisplayName : undefined;
-        const symbol = createSymbolForElement(element, parentElementName, fileName);
+    Structure.walkCodeElements(elements, (element) => {
+        const symbol = createSymbolForElement(element, filename);
 
         results.push(symbol);
     });
 
     return results;
 }
 
-function createSymbolForElement(element: Structure.CodeElement, parentElementName: string, fileName: string): vscode.SymbolInformation {
-    const range = element.Ranges[SymbolRangeNames.Full];
-
-    const vscodeRange = new vscode.Range(
-        range.Start.Line - 1, range.Start.Column - 1, range.End.Line - 1, range.End.Column - 1
-    );
-
-    return new vscode.SymbolInformation(
-        element.DisplayName,
-        toSymbolKind(element.Kind),
-        parentElementName,
-        new vscode.Location(vscode.Uri.file(fileName), vscodeRange)
-    );
+function createSymbolForElement(element: Structure.CodeElement, filename: string): vscode.DocumentSymbol {
+    const fullRange = element.Ranges[SymbolRangeNames.Full];
+    const nameRange = element.Ranges[SymbolRangeNames.Name];
+    
+    return new vscode.DocumentSymbol(element.DisplayName, filename, toSymbolKind(element.Kind), toRange3(fullRange), toRange3(nameRange));
```

Comment:
Is this still hierarchical? Do we need to populate the [`children`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L2569-L2572) property on [`DocumentSymbol`](https://code.visualstudio.com/docs/extensionAPI/vscode-api#DocumentSymbol)? Or, does that happen automatically passed on the ranges that are passed in?